src/Kunstmaan/AdminBundle/Tests/Helper/AdminRouteHelperTest.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace Kunstmaan\AdminBundle\Tests\Helper;
+
+use Kunstmaan\AdminBundle\Helper\AdminRouteHelper;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Kunstmaan\NodeBundle\Router\SlugRouter;
+
+class AdminRouteHelperTest extends \PHPUnit_Framework_TestCase
+{
+    protected static $ADMIN_KEY = "admin";
+    protected static $NON_ADMIN_URL = "/blog/page/1";
+    protected static $ADMIN_URL = "/admin/blog/page/1";
+    protected static $PREVIEW_ADMIN_URL = "/admin/preview/blog/page/1";
+
+    /**
+     * @var AdminRouteHelper
+     */
+    protected $adminRouteHelper;
+
+    /**
+     * Sets up the fixture, for example, opens a network connection.
+     * This method is called before a test is executed.
+     */
+    protected function setUp()
+    {
+        $this->adminRouteHelper = $this->getAdminRouteHelper();
+    }
+
+    /**
+     * Tears down the fixture, for example, closes a network connection.
+     * This method is called after a test is executed.
+     */
+    protected function tearDown()
+    {
+    }
+
+    /**
+     * @covers Kunstmaan\AdminBundle\Helper\AdminRouteHelper::isAdminRoute
+     */
+    public function testIsAdminRouteReturnsTrueWhenAdminUrl()
+    {
+        $result = $this->adminRouteHelper->isAdminRoute(self::$ADMIN_URL);
+
+        $this->assertTrue($result);
+    }
+
+    /**
+     * @covers Kunstmaan\AdminBundle\Helper\AdminRouteHelper::isAdminRoute
+     */
+    public function testIsAdminRouteReturnsFalseWhenFrontendUrl()
+    {
+        $result = $this->adminRouteHelper->isAdminRoute(self::$NON_ADMIN_URL);
+
+        $this->assertFalse($result);
+    }
+
+    /**
+     * @covers Kunstmaan\AdminBundle\Helper\AdminRouteHelper::isAdminRoute
+     */
+    public function testIsAdminRouteReturnsFalseWhenPreviewUrl()
+    {
+        $requestStack = $this->getMock('Symfony\Component\HttpFoundation\RequestStack');
```

Comment:
well, in fact you don’t even need to mock the `RequestStack`, just use it directly. 

```php
$requestStack = new RequestStack;
$requestStack->push($this->getPreviewRequest());
```